doc/api/esm.md

```diff
@@ -86,8 +86,22 @@ When loaded via `import` these modules will provide a single `default` export
 representing the value of `module.exports` at the time they finished evaluating.
 
 ```js
-import fs from 'fs';
-fs.readFile('./foo.txt', (err, body) => {
+// foo.js
+module.exports = { one: 1 };
+```
+
+```js
+// bar.js
+import foo from './foo.js';
+foo.one === 1; // true
+```
+
+Builtin modules such as fs will provide the above with the addition of their
+public documented api as named exports.
```

Comment:
I'd also like to see some discussion on userland monkeypatching through CJS, specifically that ESM exports will never change, but these exported functions may then call the monkeypatched method internally since they are still implemented in CJS.

Reply:
The hope, i think, is to prevent userland monkeypatching of CJS modules to have any impact on ESM imports of core modules.

Reply:
@ljharb Yeah, I understand. Well that isn't really possible at this moment, and in fact there are specific workarounds in Node.js to allow for monkeypatching, so I'd rather document it.